### ISSUE TITLE
Test Cases Name restrictions will match any result

### ISSUE THREAD

**Issue by giuseppeverduciALMA** [buildingSMART/IDS#169]
In the following test cases:

https://github.com/buildingSMART/IDS/blob/master/Documentation/testcases-attribute.md#pass-name-restrictions-will-match-any-result-13
https://github.com/buildingSMART/IDS/blob/master/Documentation/testcases-attribute.md#pass-name-restrictions-will-match-any-result-23
https://github.com/buildingSMART/IDS/blob/master/Documentation/testcases-attribute.md#pass-name-restrictions-will-match-any-result-33

according to me "any" should be "all".

The restriction filters out many attributes (name and description), and all attributes should respect the meaning "The attribute must exist and have a non-empty value" (https://github.com/buildingSMART/IDS/blob/master/Documentation/attribute-facet.md#parameters).

**Comment by aothms** [comment]
I think in nearly all cases we have opted for any instead of all. E.g a concrete wall is a wall where at least one layer is made out of concrete, not all layers. Or, I need property X in pset Y or Z, not in both Y and Z. Attributes are subtle different, because the schema defines their names and cardinalities, but still I'd propose to keep the behaviour to any for consistency sake.

**Comment by giuseppeverduciALMA** [comment]
I agree on uniformity of behaviour, but we must consider that there are 2 cases:
- tests of values: material, classification, value of a property and value of an attribute
- tests of container of values: property and attribute

in the first case, I agree that "any" is OK
in the second case, however, "all" must be considered

first case example:
property must have value "one|two"

second case example:
exist properties with name ".*test.*"
